subcommands/waves/init.go

```diff
@@ -0,0 +1,160 @@
+package waves
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/docker/go/canonical/json"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	tuf "github.com/theupdateframework/notary/tuf/data"
+)
+
+func init() {
+	initCmd := &cobra.Command{
+		Use:   "init <wave> <version> [<tag>]",
+		Short: "Create a new wave for a given targets version.",
+		Long: `Create a new wave for a given targets version.",
+Optionally, provide a new tag to use for these targets ('master' by default).
+In any case, original tags of these targets set before creating a wave are ignored.
+
+This command only initializes a wave, but does not provision its updates to devices.
+Use a "fioctl wave rollout <wave> <group>" to trigger updates of this wave to one group.
+Use a "fioctl wave promote <wave>" to update all devices (make it globally available).`,
+		Run:  doInitWave,
+		Args: cobra.RangeArgs(2, 3),
+	}
+	cmd.AddCommand(initCmd)
+	initCmd.Flags().IntP("expires-days", "e", 0, `Role expiration in days; default 90.
+The same expiration will be used when a wave is promoted.`)
```

Comment:
@ricardosalveti - ATS tooling does 1 year for the target expiration. The notary tooling seems a little more conservative. What do you think?

Reply:
I was waiting for this comment, cause I'm not sure on this default expiration interval.

Currently, I've found the following expiration policies:
1. `theupdateframework/notary/const.go` used by a notary server defaults are: root - 10 years, targets - 3 years, snapshot - 3 years, timestamp - 14 days.
2. `theupdateframework/notary/tuf/data/types.go` recommended TUF defaults are: root - 1 year, targets - 90 days, snapshot - 7 days, timestamp - 1 day.
3. `advancedtelematic/ota-tuf/.../libtuf_server/repo/server/SignedRoleGeneration.scala` uses a 31 days default for targets, snapshot, and timestamp roles (keeps them in sync).
4. Current ota-lite implementation for condensed targets uses 30 days expiration for targets, and then via tuf_reposerver the same expiration is set for snapshot/timestamp roles.

My thinking on this is the following:
Clearly, notary server many-years expiration policy is way too conservative.  Still, a one month expiration in use for CI targets is too aggressive for production, where release cycles tend to take a longer time.  Thus, I voluntarily used a 90 days expiration recommended by TUF for targets role, though I'm thinking of a longer default, like 1 year.
A problem I see with "too short" expiration is that a user would need to re-sign production targets in order to keep them valid in case if their release cycle is longer than an expiration interval.  I think we can add a `fioctl targets sign` command for this, but that simply requires too many moves from a customer side.

Still, for snapshot/timestamp roles I'd use a rather aggressive expiration policy in production e.g. 2-4 weeks for a snapshot roles and 3-7 days for a timetamp role.  Still, I find a 1 day expiration for timestamp too aggressive, because in case of a re-signer daemon failure we might end up with invalidated timestamp roles for all customers within a day, that would be a maintenance nightmare.

That said, my current plan is this: 90 days expiration for targets, 14 days for snapshot, 3 days for timestamp;   though I may easily change my mind to be more conservative like this: 1 year for targets, 30 days for snapshot, 7 days for timestamp.

@doanac @ricardosalveti @mike-sul what do you think ?

Reply:
Resolved offline:
Will use 1 year default for targets expiration (configurable), snapshot expiration equal to targets, timestamp expiration 7 days.